web/src/beta/components/Resizable/hooks.ts

```diff
@@ -0,0 +1,118 @@
+    if (which === 3) return;
+
+    return { x, y };
+  }
+
+  if (nativeEvent instanceof TouchEvent) {
+    const touch = nativeEvent.touches[0];
+    const { clientX: x, clientY: y } = touch;
+
+    return { x, y };
+  }
+
+  return;
+};
+
+const getDelta = (direction: Direction, deltaX: number, deltaY: number) =>
+  direction === "vertical" ? deltaX : deltaY;
+
+const getSize = (size: number, delta: number, minSize?: number, maxSize?: number) => {
+  if (minSize !== undefined && size + delta < minSize) return minSize;
+  if (maxSize !== undefined && size + delta > maxSize) return maxSize;
+  return size + delta;
+};
+
+export default (
+  direction: Direction,
+  gutter: Gutter,
+  initialSize: number,
+  minSize?: number,
+  maxSize?: number,
+) => {
+  const [isResizing, setIsResizing] = useState(false);
+  const [size, setSize] = useState(initialSize);
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+
+  const onResizeStart = useCallback(
+    (e: React.MouseEvent<HTMLDivElement> | React.TouchEvent<HTMLDivElement>) => {
+      const position = getPositionFromEvent(e);
+      if (!position) return;
+
+      setIsResizing(true);
+      setPosition(position);
+    },
+    [],
+  );
+
+  const onResize = useCallback(
+    (e: MouseEvent | TouchEvent) => {
+      if (!isResizing) return;
+
+      const { clientX: x, clientY: y } = e instanceof MouseEvent ? e : e.touches[0];
+      const deltaX = gutter === "start" ? position.x - x : x - position.x;
+      const deltaY = gutter === "start" ? position.y - y : y - position.y;
+      const delta = getDelta(direction, deltaX, deltaY);
+
+      setSize(getSize(size, delta, minSize, maxSize));
+      setPosition({ x, y });
+    },
+    [isResizing, position, direction, gutter, size, minSize, maxSize],
+  );
+
+  let unbind = () => {}; // eslint-disable-line @typescript-eslint/no-empty-function
+
+  const onResizeEnd = useCallback(() => {
+    if (!isResizing) return;
+
+    setIsResizing(false);
+    setPosition({ x: 0, y: 0 });
+    unbind();
+  }, [isResizing]);
+
+  const bindEventListeners = useCallback(() => {
+    if (typeof window === "undefined") return;
+    window.addEventListener("mouseup", onResizeEnd);
+    window.addEventListener("mousemove", onResize);
+    window.addEventListener("mouseleave", onResizeEnd);
+    window.addEventListener("touchmove", onResize);
+    window.addEventListener("touchend", onResizeEnd);
+  }, [onResize, onResizeEnd]);
+
+  const unbindEventListeners = useCallback(() => {
+    if (typeof window === "undefined") return;
+    window.removeEventListener("mouseup", onResizeEnd);
+    window.removeEventListener("mousemove", onResize);
+    window.removeEventListener("mouseleave", onResizeEnd);
+    window.removeEventListener("touchmove", onResize);
+    window.removeEventListener("touchend", onResizeEnd);
+  }, [onResize, onResizeEnd]);
+
+  unbind = unbindEventListeners;
```

Comment:
Why do we need this line?

Reply:
I just wanted to use the same resizable as the classic and want to change only gutters size.
So didn't check it.

Took a look at it a little, but this line seems not to be able to delete easily. It might avoid an infinite loop of something.
Could I deal with it on another PR later?